### DISCUSSION TITLE
Devices frequently reporting offline for a moment

### DISCUSSION THREAD

**Discussion by srudin** [tsightler/ring-mqtt#894] [Support Q&A]
I had my installation with HASS, Mosquitto and RingMqtt (everything on Docker) finished several months ago and everything was running fine for a while. Starting Aug 15th at 8:41pm local time I am seeing gaps in the Ring alarm history:

![image](https://github.com/user-attachments/assets/eb3e0719-3898-48b7-991e-50355750b4cf)

The problem is that I have a bunch of automations based on checks when the state of an entity changes (e.g. when Ring alarm becomes “Armed” or “Unarmed”) and because these interruptions turn all respective entities to state “Unknown” and then back this always triggers the related automations - or delays the triggering when I would want it in case of an interruption. So this really needs fixing.

On Aug 15th around 8:30pm I have installed the 4.3.1.R752 Asustor update so it seems obvious that this caused the issues. I found out that the gaps in HASS are caused by RingMqtt reporting an "offline" status for all devices:

![image](https://github.com/user-attachments/assets/3e9d1814-6cf8-4ead-90a2-94127545bad1)

However, nothing really went offline in that moment and in the Ring app all devices look fine.

I am aware that based on these facts it is rather Asustor that should do support here and not RingMqtt. Because of this I have already created a post in the Asustor forum (although my expectations for support are rather low there).

Still I would like to ask if someone has seen this behavior already? What exactly is causing RingMqtt to report these offline status? A network exception? Any way to turn on logging for RingMqtt and get more information on what is going wrong?

**Comment by tsightler** [comment]
I have no idea what Asustor is, but I'd bet good money they've been bitten by the same network issue which impacted Home Assistant OS 13 when running as a Proxmox VM (maybe any Linux based network virtualization).  There was a kernel bug that caused lots of dropped packets, especially small packets and it produced behavior exactly like this.

Overall, ring-mqtt is not magical, it only reports what it sees.  It creates a basic websocket connection using standard libraries and all alarm devices are monitored over this single connection (same is Ring app).  For "online/offline", this is an indication if the state for this entity is currently available or not and ring-mqtt simply monitors the status of the websocket and, if the websocket is connected, then state is available, otherwise it is not, simple as that.  Anything that interrupts the websocket will cause devices to go offline.

Of course there is logging, debug logging options are covered in the documentation, although, most likely you would only see messages that the websocket was closed or re-established so I'm not sure how much that would help.

**Reply by srudin** [reply]
Ok thanks for the quick answer, I will add this to the Asustor support issue.
(Btw Asustor is just the name of NAS products manufactured by Asus. They are just small Linux based computers.)
